src/test/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/ComparisonServiceTest.kt

```diff
@@ -0,0 +1,136 @@
+import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.repository.ComparisonPersonRepository
+import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.repository.ComparisonRepository
+import java.time.LocalDateTime
+import java.util.UUID
+
+@ExtendWith(MockitoExtension::class)
+class ComparisonServiceTest : IntegrationTestBase() {
+
+  private val prisonService = mock<PrisonService>()
+
+  @Autowired
+  lateinit var comparisonService: ComparisonService
+
+  @Autowired
+  lateinit var objectMapper: ObjectMapper
+
+  @Test
+  fun `A Comparison is created when create is called`() {
+    SecurityContextHolder.setContext(
+      SecurityContextImpl(AuthAwareAuthenticationToken(FAKE_TOKEN, USERNAME, emptyList())),
+    )
+
+    val comparisonInput = ComparisonInput(criteria = objectMapper.createObjectNode(), manualInput = false, prison = null)
+    val comparison = comparisonService.create(comparisonInput)
+    Assertions.assertEquals(comparison.manualInput, false)
+    Assertions.assertEquals(comparison.prison, null)
+  }
+
+  @Test
+  fun `A Comparison is created when create is called with manual input`() {
+    SecurityContextHolder.setContext(
+      SecurityContextImpl(AuthAwareAuthenticationToken(FAKE_TOKEN, USERNAME, emptyList())),
+    )
+
+    val comparisonInput = ComparisonInput(criteria = objectMapper.createObjectNode(), manualInput = true, prison = "ABC")
+    val comparison = comparisonService.create(comparisonInput)
+    Assertions.assertEquals(comparison.manualInput, true)
+    Assertions.assertEquals(comparison.prison, "ABC")
+  }
+
+  @Test
+  fun `Get a list of comparisons`() {
+    val prisonService = mock<PrisonService>()
+    val comparisonRepository = mock<ComparisonRepository>()
+    val comparisonPersonRepository = mock<ComparisonPersonRepository>()
+    val serviceUserService = mock<ServiceUserService>()
+
+    // Creating an alternative comparison service so that we can mock the prisonService response
+    val alternativeComparisonService = ComparisonService(comparisonRepository, prisonService, serviceUserService, comparisonPersonRepository)
+
+    Mockito.`when`(prisonService.getCurrentUserPrisonsList()).thenReturn(listOf("ABC"))
+    Mockito.`when`(comparisonRepository.findAllByManualInputAndPrisonIsIn(any(), any())).thenReturn(
+      listOf(
+        Comparison(
+          1,
+          UUID.randomUUID(),
+          "ABCD1234",
+          objectMapper.createObjectNode(),
+          "ABC",
+          false,
+          LocalDateTime.now(),
+          "JOEL",
+        ),
+      ),
+    )
+
+    SecurityContextHolder.setContext(
+      SecurityContextImpl(AuthAwareAuthenticationToken(FAKE_TOKEN, USERNAME, emptyList())),
+    )
+
+    val comparisonList = alternativeComparisonService.listComparisons()
+
+    Assertions.assertEquals(comparisonList.size, 1)
+  }
+
+  @Test
+  fun `Get a list of manual comparisons`() {
+    SecurityContextHolder.setContext(
```

Comment:
Just intrigued if your new shiny encapsulation could be used as a mock - I haven't followed this bit of code all the way through so I could be wrong.